refactor!: move cogs to the main library (#696)
nextcord/nextcord: docs/ext/commands/cogs.rst

Cogs
====

There comes a point in your bot's development when you want to organize a collection of commands, listeners, and some state into one class. Cogs allow you to do just that.

The gist:

- Each cog is a Python class that subclasses :class:`.commands.Cog`.

> - Every command is marked with the :func:`.commands.command` decorator.

Reviewer: > - Every command is marked with the :func:`.commands.command` decorator.

This should probably be documented somewhere, it seems like it's not.

Also, maybe regular Cog docs should link to commands.Cog telling people to look there for ext commands. Because of the naming, I could see it confusing some people.

.. code-block:: python3

    bot.add_cog(Greetings(bot))

This binds the cog to the bot, adding all commands and listeners to the bot automatically.

Note that we reference the cog by name, which we can override through :ref:`ext_commands_cogs_meta_options`. So if we ever want to remove the cog eventually, we would have to do the following.

.. code-block:: python3

    bot.remove_cog('Greetings')

Using Cogs
----------

Just as we remove a cog by its name, we can also retrieve it by its name as well. This allows us to use a cog as an inter-command communication protocol to share data. For example:

.. code-block:: python3
    :emphasize-lines: 22,24

    class Economy(commands.Cog):
        ...

        async def withdraw_money(self, member, money):
            # implementation here
            ...

        async def deposit_money(self, member, money):
            # implementation here
            ...

    class Gambling(commands.Cog):
        def __init__(self, bot):
            self.bot = bot

        def coinflip(self):
            return random.randint(0, 1)

>         @commands.command()

Reviewer 2: There is already a quick example that contains a `commands.Command`.

- :meth:`.Cog.bot_check`
- :meth:`.Cog.bot_check_once`

You can visit the reference to get more detail and see additional special methods in :ref:`cogs`.

.. _ext_commands_cogs_meta_options:

Meta Options
--------------

Meta options are exactly like what they are in :class:`nextcord.Cog`, however there is an additional option.

> If you would like to see what this extra option is, see the documentation of :class:`.CogMeta`.

Reviewer: I think the wording here could be improved
```suggestion
For more information, see the documentation of :class:`.CogMeta`.
```
